app/services/done_index_job.rb

```diff
@@ -1,57 +1,64 @@
 class DoneIndexJob < BaseIndexJob
-  attr_reader :results
+  attr_reader :status, :ran_job, :es_stats, :time_took, :message
 
-  def self.build_object(body:, when_completed_proc:)
-    new(results: body[:results],
-        book_version_id: body[:book_version_id],
-        indexing_strategy_name: body[:indexing_strategy_name],
-        when_completed_proc: when_completed_proc)
+  STATUS = [
+    STATUS_SUCCESSFUL                = "successful",
+    STATUS_INVALID_INDEXING_STRATEGY = "invalid indexing strategy",
+    STATUS_OTHER_ERROR               = "other error"
+  ]
+
+  def self.build_object(params:, cleanup_after_call:)
+    ran_job_params = JSON.parse(params[:ran_job]).with_indifferent_access
+    ran_job_type = ran_job_params[:type].constantize
+    ran_job = ran_job_type.build_object(params: ran_job_params)
+
+    new(status: params[:status],
+        es_stats: params[:es_stats],
+        time_took: params[:time_took],
+        message: params[:message],
+        ran_job: ran_job)
   end
 
-  def initialize(results:, book_version_id:, indexing_strategy_name:, when_completed_proc: nil)
-    super(book_version_id: book_version_id,
-          indexing_strategy_name: indexing_strategy_name,
-          when_completed_proc: when_completed_proc)
+  def initialize(status: STATUS_SUCCESSFUL,
+                 es_stats: nil,
+                 time_took: nil,
+                 ran_job: nil,
+                 message: nil,
+                 cleanup_after_call: nil)
+    super(cleanup_after_call: cleanup_after_call)
```

Comment:
if you pass `book_version_id` and `index_strategy_name` here with the values from `ran_job` then this done job's `attr_reader` for those fields will return something and you won't have to override `find_associated_book_index` below.